lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/BaseRuleBasedConnectorAccessControlTest.java

```diff
@@ -0,0 +1,677 @@
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, AGGREGATE, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, ALICE.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, SCALAR, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, ALICE.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, WINDOW, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, ALICE.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, AGGREGATE, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, BOB.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, SCALAR, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, BOB.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, WINDOW, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, BOB.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, AGGREGATE, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, CHARLIE.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, SCALAR, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, CHARLIE.getIdentity().getUser()), true);
+        accessControl.checkCanGrantExecuteFunctionPrivilege(BOB, WINDOW, new SchemaRoutineName("any", "some_function"), new TrinoPrincipal(PrincipalType.USER, CHARLIE.getIdentity().getUser()), true);
+    }
+
+    @Test
+    public void testEverythingImplemented()
+            throws NoSuchMethodException
+    {
+        assertAllMethodsOverridden(ConnectorAccessControl.class, RuleBasedAccessControl.class);
+    }
+
+    @Test
+    public void testInvalidRules()
+    {
+        assertThatThrownBy(() -> createAccessControl("invalid.json"))
+                .hasMessageContaining(getErrorMessagePrefix());
+    }
+
+    protected ConnectorAccessControl createAccessControl(Module accessControlModule, Map<String, String> configProperties)
+    {
+        Bootstrap bootstrap = new Bootstrap(new CatalogNameModule("test_catalog"), accessControlModule);
+
+        Injector injector = bootstrap
+                .doNotInitializeLogging()
+                .setRequiredConfigurationProperties(configProperties)
+                .initialize();
+
+        return injector.getInstance(ConnectorAccessControl.class);
+    }
+
+    protected String getResourcePath(String resourceName)
```

Comment:
Is it used only in `TestRestBasedAccessControl` in case can we move this method to that class ?

Reply:
moved